core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -136,15 +181,92 @@ abstract class AccumulatorV2[IN, OUT] extends Serializable {
   def reset(): Unit
 
   /**
+   * Takes the inputs and accumulates. e.g. it can be a simple `+=` for counter accumulator.
+   * Developers should extend addImpl to customize the adding functionality.
+   */
+  final def add(v: IN): Unit = {
```

Comment:
Yah there isn't a way to do this for everyone without breaking binary compatibility. Given that the AccumulatorV2 API was added in Spark2.0 though my only argument is that I don't expect anyone would be using this yet (given the current usability of accumulators) and maybe it would be ok if we mention it in the release notes?

But if you think the best path forward is subclassing that makes sense too.